Design/Mantid_As_a _SciPy_Styled_Package.md

```diff
@@ -0,0 +1,259 @@
+* A list of Algorithms within this module
+* A list of Fit functions within this module
+* Other objects
+*  An example or two of things that are possible using this module with python examples that are doc tested
+
+Each of the lists will be generated by a directive, and should not appear at all if there is nothing in the list.
+?? is it possible to extract the short module description from a docstring? ?? - If so all of the top can be replaced with a single directive.
+
+### Algorithm and Fit Function Pages ###
+
+These will remain as is, except that the category links will be redirected to the module pages.
+
+C++ Library Structure
+---------------------
+
+In order to make the C++ library easy to navigate we will rearrange the library to match the module structure of the python modules.  This will increase the number of libraries we have, but will help split out some of those that have become bloated, such as Algorithms and DataHandling.
+
+?? Do we want separate C++ libaries for sub-modules such as mantid.math.axes or should be stop at a certain level and gather all the mantid.math into a single library? ??
+
+If we decide to allow objects (such as algorithms) to be aliased in other python modules then we will need to define a rule of where the C++ algorithm will reside, we should use the first entry in the algorithm Category to be where the real code exists and the rest are aliases.
+
+Low Level operations
+--------------------
+
+There are a number of low level operations that would be good to expose for use in Python on numpy arrays with the associated workspace structures.  This would allow other scripts or indeed algorithms within mantid to make use of the central calculations in situations when building a workspace is not appropriate.
+
+Deciding which operations to expose like this should be constrained by the following guidelines:
+
+1. We should not replicate functionality that is directly available in NumPy, SciPy etc.
+1. All operations should work with inputs and outputs that are exported through simple python object and NumPy arrays.
+1. We should only consider heavily used core operations for exposing in advance of a specific user or developer need.
+
+Examples that we could expose:
+``` python 
+# For simple conversions of an array of values to another neutronic unit
+result_numpy_array = mantid.api.functions.convertUnits(input_numpy_array, target_unit, l1, l2, two_theta, energy_mode, e_fixed)
+# For rebinning a simple pair of matched arrays
+(data_numpy_array, boundaries_numpy_array) = mantid.api.functions.rebin(data_numpy_array, boundaries_numpy_array, rebin_params_list)
+```
+
+With the convertUnits function we could rewrite TOF_Converter and remove a lot of the custom unit conversion code.
+
+Questions
+=========
+
+1. Mantid.IO should we split out NeXus, ISAW etc?
+1. Should mantid.simulations be mantid.neutron.simulations ?
+1. If you have imported a sub selection of packages of IO, what does that mean for Load?
```

Comment:
Good point. It depends how context sensitive we want `Load` to be. If they do `mantid.io.nexus.Load` then it would only load the nexus loading algorithms and `Load` would not be able to load `.raw` files for example but I don't think I would expect it to as it's been imported from the `nexus` package - this could get too complicated though...

I think this is down at the detail level but we need to think about what is the easiest for our users.